src/Manager.hpp

```diff
@@ -315,6 +318,28 @@ inline std::vector<Entity*> ECS::EntitiesWith(Ts&& ...types)
 	return requestedEntities;
 }
 
+template<typename T>
+inline bool ECS::HasComponent(Entity entity, T component)
+{
+	std::string componentName = this->GetComponentName(component);
+	if (!this->ComponentIsRegistered(componentName))
+	{
+		throw ComponentNotRegisteredException(componentName);
+	}
+
+	int componentFlag = componentIndex[componentName];
+
+	for (Entity& e : this->entities)
+	{
+		if (e.UUID == entity.UUID)
+		{
+			return bitfield::Has(e.bitfield, componentFlag);
+		}
+	}
+
+	throw std::runtime_error("Failed to find entity to remove component from");
+}
```

Comment:
Untested, but this function is effectively a duplicate of `GetComponent(...)`. We could probably leverage that and condense this guy down to 
```
{
    return this->GetComponent(entity, component) != nullptr ? true : false;
}
```
Bonus on that possibility: `RemoveComponent` could probably leverage `GetComponent` as well since that also has to look for the entity first, which would help remove some of the duplication in that function.

Reply:
Did the GetComponent trick. Added a new method to get the ComponentName and ComponentFlag in one shot as well.

iterating the entire entity list is a bit more complex (pointer v no pointer) so I am leaving it alone for now.